main.ts

```diff
@@ -72,6 +76,44 @@ class Translator {
     this.visit(expr);
   }
 
+  /** Searches for super() calls and emits an initializer for it if found. */
+  maybeEmitInitializerForSuperCall(body: ts.Block) {
+    if (!body || body.statements.length === 0) return;
+
+    var findSuperCalls = (stmt: ts.Statement) => {
+      if (stmt.kind !== ts.SyntaxKind.ExpressionStatement) return;
+      var nestedExpr = (<ts.ExpressionStatement>stmt).expression;
+      if (nestedExpr.kind !== ts.SyntaxKind.CallExpression) return;
+      var callExpr = <ts.CallExpression>nestedExpr;
+      if (callExpr.expression.kind !== ts.SyntaxKind.SuperKeyword) return;
+
+      this.emit(': super (');
+      this.visitList(callExpr.arguments);
+      this.emit(')');
+    };
+
+    body.statements.forEach(findSuperCalls);
+  }
+
+  /**
+   * Checks whether `callExpr` is a super() call that should be ignored because it was already
+   * handled by `maybeEmitSuperInitializer` above.
+   */
+  isHandledSuperCall(callExpr: ts.CallExpression): boolean {
```

Comment:
There's two things that have to happen:
1) when you hit the ctor, you find the super call and move it to the initializer
2) when you hit a super() call, you have to find if it should be removed or error out

(2) has to happen independently of (1), as you could find a super call anywhere in the tree.

So I think this is the best we can do here.